src/_pytest/main.py

```diff
@@ -185,10 +185,12 @@ def wrap_session(config, doit):
             session.exitstatus = EXIT_TESTSFAILED
         except KeyboardInterrupt:
             excinfo = _pytest._code.ExceptionInfo()
-            if initstate < 2 and isinstance(excinfo.value, exit.Exception):
+            exitstatus = EXIT_INTERRUPTED
+            if initstate <= 2 and isinstance(excinfo.value, exit.Exception):
```

Comment:
I had to change the the `<` to `<=` to make the statement true. Can you tell me why @nicoddemus ?

Reply:
Hmm not really, it should not affect this at all...

Reply:
I suggest we handle `None` here like this:

```python
            exitstatus = EXIT_INTERRUPTED
            if initstate <= 2 and isinstance(excinfo.value, exit.Exception):
                sys.stderr.write("{}: {}\n".format(excinfo.typename, excinfo.value.msg))
                if excinfo.value.returncode is not None:
                    exitstatus = excinfo.value.returncode
```